core/src/oauth/credential.rs

```diff
@@ -0,0 +1,113 @@
+use core::fmt;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::oauth::store::OAuthStore;
+use crate::utils::{self, ParseError};
+use anyhow::Result;
+
+#[derive(Debug, Clone, Copy, Serialize, PartialEq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum CredentialProvider {
+    Snowflake,
+}
+
+impl fmt::Display for CredentialProvider {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            serde_json::to_string(&self).unwrap().trim_matches('"')
+        )
+    }
+}
+
+impl FromStr for CredentialProvider {
+    type Err = ParseError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match serde_json::from_str(&format!("\"{}\"", s)) {
+            Ok(v) => Ok(v),
+            Err(_) => Err(ParseError::new()),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Clone)]
+pub struct Credential {
+    credential_id: String,
+    created: u64,
+    provider: CredentialProvider,
+    credentials: serde_json::Map<String, serde_json::Value>,
+}
```

Comment:
We should have encrypted_content/content/value here + sealing / unsealing should be done in that file not the store.

Reply:
Not against it at all but can you explain your reasonning ?

I felt it made sense to have the encryption / decryption to / from storage be done in the store file. We read data from the db and we return the content, the fact that it's encrypted in the db is a detail that the rest of the code do not need to be aware of, ihmo.

However, I see the inconsistency with connexion. I can move both to the "new pattern" if you agree that it's better.

Reply:
Main arguments are:
- separation of concern (having the crypto code not mixed with DB management)
- concistency with the rest of the service codebase

Reply:
I think encryption should be in a very visible and contained place. store.rs does not really fit that IMHO

Reply:
I updated the code to match what I think you had in mind. But now the unseal (the action not the actual code) happens in oauth.rs before sending the response which feel weird. Did I misunderstood ?